pennylane/fourier/reconstruct.py

```diff
@@ -0,0 +1,295 @@
+                _shifts = shifts.get(arg_name)
+                if _shifts is not None:
+                    _shifts = _shifts.get(par_idx)
+
+                _spectrum = spectra[arg_name][par_idx]
+                # Determine whether fun_at_zero is needed
+                if _shifts is None or any(np.isclose(_shifts, 0.0, rtol=0, atol=atol)):
+                    need_fun_at_zero = True
+
+                # Store job; fun_at_zero missing
+                if len(_spectrum) > 1:
+                    _jobs[par_idx] = {"shifts": _shifts, "spectrum": _spectrum}
+                else:
+                    _jobs[par_idx] = None
+
+            jobs[arg_name] = _jobs
+
+    else:
+        jobs = {}
+        need_fun_at_zero = True
+        if ids is None:
+            # ids = nums_frequency.keys()
+            ids = {outer_key: inner_dict.keys() for outer_key, inner_dict in nums_frequency.items()}
+        recon_fn = _reconstruct_equ
+
+        for arg_name, inner_dict in ids.items():
+            _jobs = {}
+
+            for par_idx in inner_dict:
+                _num_frequency = nums_frequency[arg_name][par_idx]
+                # Store job; fun_at_zero missing
+                jobs[par_idx] = {"num_frequency": _num_frequency} if _num_frequency > 0 else None
+
+            jobs[arg_name] = _jobs
+
+    return ids, recon_fn, jobs, need_fun_at_zero
+
+
+def reconstruct(qnode, ids=None, nums_frequency=None, spectra=None, shifts=None):
+    r"""Reconstruct univariate restrictions of an expectation value QNode.
+
+    Args:
+        qnode (pennylane.QNode): Quantum node to be reconstructed, representing a
+            circuit that outputs an expectation value.
+        ids (list): Indices for the QNode parameters with respect to which
+            the QNode should be reconstructed as a univariate function.
+            Each entry of the list should be a valid key for ``nums_frequency`` if
+            it is provided or ``spectra``. Alternatively, a single valid key may be provided.
+            If ``None``, all keys of ``nums_frequency``/``spectra`` are considered.
+        nums_frequency (pennylane.ArgMap[int]): Numbers of integer frequencies -- and biggest
+            frequency -- per QNode parameter. If the frequencies are not contiguous integers,
+            the argument ``spectra`` should be used. Takes precedence over ``spectra`` and
+            uses equidistant shifts.
+        spectra (pennylane.ArgMap[list[float]]): Frequency spectra per QNode parameter.
+            Ignored if ``nums_frequency!=None``.
+        shifts (pennylane.ArgMap[list[float]]): Shift angles for computing the reconstruction per
+            QNode parameter. Ignored if ``nums_frequency!=None``.
+    Returns:
+        function: Function which accepts the same arguments as the QNode.
+            When called, this function will return a :class:`~.pennylane.argmap.ArgMap`
+            that contains the univariate reconstructions per QNode parameter.
```

Comment:
```suggestion
            QNode parameter. Ignored if ``nums_frequency!=None``.

    Returns:
        function: Function which accepts the same arguments as the QNode.
        When called, this function will return a :class:`~.pennylane.argmap.ArgMap`
        that contains the univariate reconstructions per QNode parameter.
```

Reply:
Wait, is the mention of ArgMap outdated?

Reply:
Yes, sorry :D